code/game/gamemodes/cult/blood_magic.dm

```diff
@@ -0,0 +1,816 @@
+				var/obj/item/organ/external/affecting = H.get_organ(dam_zone)
+				H.apply_damage(5 - rune * 3, BRUTE, affecting)
+			else
+				H.bleed(20 - rune * 12)
+		var/datum/action/innate/cult/blood_spell/new_spell = new BS(owner)
+		new_spell.Grant(owner, src)
+		spells += new_spell
+		Positioning()
+		to_chat(owner, "<span class='warning'>Your wounds glow with power, you have prepared a [new_spell.name] invocation!</span>")
+	channeling = FALSE
+
+/datum/action/innate/cult/blood_magic/proc/remove_spell()
+	var/nullify_spell = input(owner, "Choose a spell to remove.", "Current Spells") as null|anything in spells
+	if(nullify_spell)
+		qdel(nullify_spell)
+
+/datum/action/innate/cult/blood_spell //The next generation of talismans, handles storage/creation of blood magic
+	name = "Blood Magic"
+	button_icon_state = "telerune"
+	desc = "Fear the Old Blood."
+	var/charges = 1
+	var/magic_path = null
+	var/obj/item/melee/blood_magic/hand_magic
+	var/datum/action/innate/cult/blood_magic/all_magic
+	var/base_desc //To allow for updating tooltips
+	var/invocation = "Hoi there something's wrong!"
+	var/health_cost = 0
+
+/datum/action/innate/cult/blood_spell/Grant(mob/living/owner, datum/action/innate/cult/blood_magic/BM)
+	if(health_cost)
+		desc += "<br>Deals <u>[health_cost] damage</u> to your arm per use."
+	base_desc = desc
+	desc += "<br><b><u>Has [charges] use\s remaining</u></b>."
+	all_magic = BM
+	button.ordered = FALSE
+	..()
+
+/datum/action/innate/cult/blood_spell/Remove()
+	if(all_magic)
+		all_magic.spells -= src
+	if(hand_magic)
+		qdel(hand_magic)
+		hand_magic = null
+	..()
+
+/datum/action/innate/cult/blood_spell/IsAvailable()
+	if(!iscultist(owner) || owner.incapacitated()  || !charges)
+		return FALSE
+	return ..()
+
+/datum/action/innate/cult/blood_spell/Activate()
+	if(magic_path) //If this spell flows from the hand
+		if(!hand_magic)
+			hand_magic = new magic_path(owner, src)
+			if(!owner.put_in_hands(hand_magic))
+				qdel(hand_magic)
+				hand_magic = null
+				to_chat(owner, "<span class='warning'>You have no empty hand for invoking blood magic!</span>")
+				return
+			to_chat(owner, "<span class='notice'>Your wounds glow as you invoke the [name].</span>")
+			return
+		qdel(hand_magic)
+		hand_magic = null
+		to_chat(owner, "<span class='warning'>You snuff out the spell, saving it for later.</span>")
+
+
+
+//the spell list
+
+/datum/action/innate/cult/blood_spell/stun
+	name = "Stun"
+	desc = "Empowers your hand to stun and mute a victim on contact."
+	button_icon_state = "stun"
+	magic_path = "/obj/item/melee/blood_magic/stun"
+	health_cost = 10
+
+/datum/action/innate/cult/blood_spell/teleport
+	name = "Teleport"
+	desc = "Empowers your hand to teleport yourself or another cultist to a teleport rune on contact."
+	button_icon_state = "teleport"
+	magic_path = "/obj/item/melee/blood_magic/teleport"
+	health_cost = 7
+
+/datum/action/innate/cult/blood_spell/emp
+	name = "Electromagnetic Pulse"
```

Comment:
anyway to make the cultist immune to it? While i think the suicide risk vs reward may be worth it...still

Reply:
this spell will have other cultists immune to the pulse (not yet implemented) but not the cultist himself, as you should know if your character is vulnerable to EMP or not